sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -491,7 +491,8 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
     val parsedOptions: CSVOptions = new CSVOptions(
       extraOptions.toMap,
       sparkSession.sessionState.conf.csvColumnPruning,
-      sparkSession.sessionState.conf.sessionLocalTimeZone)
+      sparkSession.sessionState.conf.sessionLocalTimeZone,
+      sparkSession.sessionState.conf.columnNameOfCorruptRecord)
```

Comment:
@MaxGekk, do you mind if I ask separate the current constructor in `CSVOptions` to, one requiring  `defaultColumnNameOfCorruptRecord` and one `defaultColumnNameOfCorruptRecord` with default value `""`, and then make some comments about when they are called?

Reply:
Do you mean:
```
  def this(
    parameters: Map[String, String],
    columnPruning: Boolean,
    defaultTimeZoneId: String,
    defaultColumnNameOfCorruptRecord: String) = {...}
```
and 
```
  def this(
    parameters: Map[String, String],
    columnPruning: Boolean,
    defaultTimeZoneId: String) = this(parameters, columnPruning, defaultTimeZoneId, "")
```
? otherwise `one defaultColumnNameOfCorruptRecord with default value ""` could be not compiled probably, but I will try.

Do you want to see that in this PR?

Reply:
Yea. Is the change a lot? I was thinking the default value constructor make it more error-prone.

Reply:
Hmm. NVM let's deal with it next time somewhere else

Reply:
Ur, I agree with @HyukjinKwon 's first opinion. This PR only fixes one instance bug. The root cause seems to be the default value `''` in the constructor.